eslint.config.mjs

```diff
@@ -0,0 +1,131 @@
+import { fixupPluginRules } from "@eslint/compat";
+import typescriptEslint from "@typescript-eslint/eslint-plugin";
+import importPlugin from "eslint-plugin-import";
+import globals from "globals";
+import tsParser from "@typescript-eslint/parser";
+import nextPlugin from "@next/eslint-plugin-next";
+import reactHooks from "eslint-plugin-react-hooks";
+
+export default [
+  {
+    ignores: [
+      "**/dist",
+      "**/build",
+      "**/node_modules",
+      "**/*.config.js",
+      "**/*.config.mjs",
+      "**/eslint.config.mjs",
+      "**/vite.config.ts",
+      "tailwind.config.js",
+      "postcss.config.js",
+      "**/*.js",
+      ".storybook/preview.ts",
+      ".storybook/main.ts",
+    ],
+  },
+  {
+    files: ["**/*.{ts,tsx}"],
+    plugins: {
+      "@typescript-eslint": fixupPluginRules(typescriptEslint),
+      "@next/next": nextPlugin,
+      import: fixupPluginRules(importPlugin),
+      "react-hooks": reactHooks,
+    },
+
+    languageOptions: {
+      globals: {
+        ...globals.browser,
+        ...globals.es2021,
+        AudioWorkletGlobalScope: "readonly",
+      },
+      parser: tsParser,
+      parserOptions: {
+        project: "./tsconfig.json",
+        ecmaVersion: "latest",
+        sourceType: "module",
+        ecmaFeatures: {
+          jsx: true,
+        },
+      },
+    },
+
+    settings: {
+      "import/resolver": {
+        typescript: {
+          project: "./tsconfig.json",
+        },
+      },
+      react: {
+        version: "detect",
+      },
+      node: {
+        extensions: [".ts", ".tsx", ".d.ts"],
+      },
+    },
+
+    rules: {
+      // Extend Next.js core web vitals rules
+      ...nextPlugin.configs["core-web-vitals"].rules,
+
+      // TypeScript specific rules
+      "@typescript-eslint/no-explicit-any": "warn",
+      "@typescript-eslint/explicit-function-return-type": "off",
+      "@typescript-eslint/no-non-null-assertion": "off",
```

Comment:
Do we have to leave this here given that our `tsconfig` allows for this?

I wouldn't want to be re-enforcing this as a choice, as i've seen it let through loads of bugs.

[See this PR spike for more info](https://github.com/climatepolicyradar/navigator-frontend/pull/377)